### DISCUSSION TITLE
Self-hosted runner stuck in "Waiting for a runner to pick up this" about 2-5 minutes

### DISCUSSION THREAD

**Discussion by liweijian** [community/community#46469] [Actions]
### Select Topic Area

Product Feedback

### Body

I'm hosting a runner on premise, at first all the tasks run fast. 

However, we got stuck in the `"Waiting for a runner to pick up this"` more than 2~5 minutes sometimes.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/19320783/217134997-95425638-a8e4-4bc7-aa6a-794912da59c7.png">

Just wondering is there any way to figure it out?

Or the only one solution is to make it autoscaling?

https://docs.github.com/en/actions/hosting-your-own-runners/autoscaling-with-self-hosted-runners

**Comment by piyushK52** [comment]
@liweijian I am stuck in the same situation. Did you find any solution?

**Reply by liweijian** [reply]
I switch to self host gitlab, it's pretty fast right now:p
